geopandas/geoseries.py

```diff
@@ -53,6 +53,30 @@ def _geoseries_constructor_with_fallback(data=None, index=None, crs=None, **kwar
         return Series(data=data, index=index, **kwargs)
 
 
+def _geoseries_expanddim(data=None, index=None, crs=None, **kwargs):
+    from geopandas import GeoDataFrame
+
+    if isinstance(data, GeoSeries):
+        # pandas default column name is 0, keep convention
+        geo_col_name = data.name if data.name is not None else 0
+    elif isinstance(data, dict):  # Dict of {name : (Geo)Series}
+        # if only one key, this is not ambiguous, if more than one
+        # i.e. from pd.concat([GeoSeries, GeoSeries]) the geom col comes
+        # from the left (same as for multiple gdfs in concat)
+        geo_col_name = list(data.keys())[0]
+    else:  # blockmanager
+        geo_col_name = data.axes[0][0]
+
+    df = GeoDataFrame(data=data, index=index, crs=crs, geometry=geo_col_name, **kwargs)
+    return df
+
+
+# pd.concat (pandas/core/reshape/concat.py) requires this for the
+# concatenation of series since pandas 1.1
+# (https://github.com/pandas-dev/pandas/commit/f9e4c8c84bcef987973f2624cc2932394c171c8c)
+_geoseries_expanddim._get_axis_number = DataFrame._get_axis_number
```

Comment:
Note this is `DataFrame._get_axis_number` rather than GeoDataFrame to avoid a circular import (but we also don't overload it so perhaps that's clearer to begin with)

Reply:
Yep. BTW, I have a PR to pandas to fix those cases: https://github.com/pandas-dev/pandas/pull/46018 (now, we will need to keep those workarounds for some time anyway, of course)